### ISSUE TITLE
aws_db instance max_allocated_storage failed on "unexpected" definition

### ISSUE THREAD

**Issue by aflatto** [terraform-providers/terraform-provider-aws#9679]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.12.6
+ provider.aws v2.16.0
+ provider.template v2.1.2
### Affected Resource(s)

* aws_db_instance

### Terraform Configuration Files

```hcl
# RDS postgresql Server
resource "aws_db_instance" "rds_server" {
  allocated_storage    = "${var.RDS_DISK}"
  max_allocated_storage = "${(var.RDS_DISK *10) }"
  identifier           = "${var.ENVIRONMENT}-rds-postgres"
  storage_type         = "gp2"
  engine               = "postgres"
  engine_version       = "9.6.11"
  instance_class       = "db.t2.micro"
  name                 = "dwOnSchedule"
  username             = "${var.RDS_USER}"
  password             = "${var.RDS_password}"
  backup_retention_period = 7
  backup_window         = "01:00-02:00"
  copy_tags_to_snapshot = true
  skip_final_snapshot  = true
  vpc_security_group_ids = ["${data.aws_security_groups.peering-sg.id}", "${data.aws_security_groups.vpn-sg.id}", "${data.aws_security_groups.office-sg.id}"  ]
  
  tags = {
      Name = "${var.ENVIRONMENT}-RDS-postgres",
      Team =  "Infra",
      Provisioner = "Terraform",
      Environment = "${var.ENVIRONMENT}"
    }
  timeouts {
      create = "10m"
      delete = "10m"
    }
  lifecycle {
    create_before_destroy = true
   }  
}```

### Expected Behavior
The resource should be created with basic storage defined and max storage defined. 

### Actual Behavior
Error: Unsupported argument

  on databases.tf line 45, in resource "aws_db_instance" "rds_server":
  45:   max_allocated_storage = "${(var.RDS_DISK *10) }"

An argument named "max_allocated_storage" is not expected here.

### Steps to Reproduce

1. `terraform plan`

**Comment by murtyjones** [comment]
I have a similar but slightly different error after upgrading to `12.6`. I have no `max_allocated_storage` argument in my RDS config, and yet get this message when `plan`ing:

```terraform
Error: unsupported attribute "max_allocated_storage"
```

**Comment by bflad** [comment]
Hi folks 👋 

The `max_allocated_storage` argument for the `aws_db_instance` resource was released in [version 2.17.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019) so you will need to upgrade your provider to that version or later to begin using the functionality. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

Note: Currently the provider documentation only shows the latest release of Terraform AWS Provider functionality. We plan on introducing versioned provider documentation later this year. 🤞

**Comment by murtyjones** [comment]
> Hi folks 👋
> 
> The `max_allocated_storage` argument for the `aws_db_instance` resource was released in [version 2.17.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019) so you will need to upgrade your provider to that version or later to begin using the functionality. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
> 
> Note: Currently the provider documentation only shows the latest release of Terraform AWS Provider functionality. We plan on introducing versioned provider documentation later this year. 🤞

I'm not using the `max_allocated_storage` attribute at all though and I'm still receiving an error.

**Comment by bflad** [comment]
@murtyjones that sounds like a similar but unrelated issue. I'd recommend opening a separate bug report so we can take a full look. 👍
